### PR TITLE
THRIFT-5237 Implement MAX_MESSAGE_SIZE and consolidate limits into a TConfiguration class(JAVA)

### PR DESCRIPTION
**Pull request by zeshuai007**
- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-5237: describe my issue"?
- [ ] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [ ] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Comment by Jens-G** [comment]
Guess we can close this?

**Comment by Jens-G** [comment]
Could you do a rebase? I'd like to look at this PR.

**Comment by zeshuai007** [comment]
Done.

File: lib/java/test/org/apache/thrift/transport/TestAutoExpandingBufferWriteTransport.java [hunk 1]
```diff
@@ -19,14 +19,16 @@
...
+import org.apache.thrift.TConfiguration;
 import org.junit.Test;
 import static org.junit.Assert.*;
 
 public class TestAutoExpandingBufferWriteTransport {
 
   @Test
   public void testIt() throws Exception {
-    AutoExpandingBufferWriteTransport t = new AutoExpandingBufferWriteTransport(1, 0);
+    AutoExpandingBufferWriteTransport t = new AutoExpandingBufferWriteTransport(new TConfiguration(), 1, 0);
```
Comment (Jens-G) [inline lib/java/test/org/apache/thrift/transport/TestAutoExpandingBufferWriteTransport.java:31]: Nut sure but .... wouldnt it be better to instantiate TConfiguration() only once and use it through the whole test case or at least inside the method?

File: lib/java/src/org/apache/thrift/protocol/TCompactProtocol.java [hunk 2]
```diff
@@ -37,868 +38,930 @@
...
+    private final static TStruct ANONYMOUS_STRUCT = new TStruct("");
+    private final static TField TSTOP = new TField("", TType.STOP, (short) 0);
+
+    private final static byte[] ttypeToCompactType = new byte[16];
+
+    static {
+        ttypeToCompactType[TType.STOP] = TType.STOP;
+        ttypeToCompactType[TType.BOOL] = Types.BOOLEAN_TRUE;
+        ttypeToCompactType[TType.BYTE] = Types.BYTE;
+        ttypeToCompactType[TType.I16] = Types.I16;
```
Comment (Jens-G) [inline lib/java/src/org/apache/thrift/protocol/TCompactProtocol.java:55]: Why all those whitespace changes? The patch is already large enough without these ;-)

File: lib/java/src/org/apache/thrift/transport/layered/TFramedTransport.java [hunk 3]
```diff
@@ -138,22 +135,24 @@ public void clear() {
...
 
-    if (size > maxLength_) {
+    if (size > getInnerTransport().getConfiguration().getMaxFrameSize()) {
       close();
       throw new TTransportException(TTransportException.CORRUPTED_DATA,
-          "Frame size (" + size + ") larger than max length (" + maxLength_ + ")!");
+          "Frame size (" + size + ") larger than max length (" + getInnerTransport().getConfiguration().getMaxFrameSize() + ")!");
     }
 
+    //updateKnownMessageSize(size);
```
Comment (Jens-G) [inline lib/java/src/org/apache/thrift/transport/layered/TFramedTransport.java:152]: Added but commented out?

File: lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java [hunk 4]
```diff
@@ -279,6 +280,10 @@ public void readFieldEnd() throws TException {}
   @Override
   public TMap readMapBegin() throws TException {
     TMap map = new TMap(readByte(), readByte(), readI32());
+
+    //
```
Comment (Jens-G) [inline lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java:284]: there are quite a few emtpy comment lines in here, is there sth missing?

**Review by Jens-G — CHANGES_REQUESTED** [review]

**Comment by Jens-G** [comment]
ping

File: lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java (hunk 4 above)
Comment (LiuXing-R) [reply lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java]: This is a redundant comment.

File: lib/java/src/org/apache/thrift/protocol/TCompactProtocol.java (hunk 2 above)
Comment (zeshuai007) [reply lib/java/src/org/apache/thrift/protocol/TCompactProtocol.java]: I'm also surprised that IDEA has modified these spaces.

File: lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java (hunk 4 above)
Comment (zeshuai007) [reply lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java]: This is a redundant comment.

**Review by Jens-G — APPROVED** [review]

**Comment by Jens-G** [comment]
+1 review LGTM, not tested. Merge?

**Comment by zeshuai007** [comment]
Thank you for reviewing this. I would like to merge it , WDYT ?

**Comment by Jens-G** [comment]
Do it.

**Comment by dcelasun** [comment]
@zeshuai007 any chance you could do something similar for Go?

**Pull request merged**